docs/docs/end-to-end-testing.md

```diff
@@ -49,10 +51,125 @@ If you want to run Cypress in Continuous Integration (CI) you have to use `cypre
+    "test:e2e:ci": "start-server-and-test develop http://localhost:8000 cy:run" // highlight-line
   }
 }
 ```
 
 Please read the [Cypress' official documentation](https://docs.cypress.io/guides/guides/continuous-integration.html) on CI if you want to know how to setup Travis or GitLab with Cypress.
+
+## Writing tests
+
+The first test is an **accessibility** test with [cypress-axe](https://github.com/avanslaars/cypress-axe).
+
+To use cypress-axe you have to install `cypress-axe` and [axe-core](https://github.com/dequelabs/axe-core). You'll also use some commands from [cypress-testing-library](https://testing-library.com/docs/cypress-testing-library/intro) to target elements easier:
+
+```bash
+npm install --save-dev cypress-axe axe-core @testing-library/cypress
+```
+
+Then you add the `cypress-axe` and `cypress-testing-library` commands in `cypress/support/commands.js`:
+
+```js:title=cypress/support/commands.js
+import "./commands"
+//highlight-start
+import "cypress-axe"
+import "@testing-library/cypress/add-commands"
+//highlight-end
+```
+
+Cypress right now will look for tests inside the `cypress/integration` folder. It makes sense to create a new folder `cypress/e2e` (end to end). You can use that folder for tests by adding the following in your `cypress.json`:
+
+```json:title=cypress.json
+{
+  "baseUrl": "http://localhost:8000/"
+  "integrationFolder": "cypress/e2e" // highlight-line
+}
+```
+
+Create a new file inside `cypress/e2e` folder and name it `a11y.test.js`.
+
+You'll use the `beforeEach` hook to run some commands before each test. After cypress loads the homepage you'll use the `checkA11y` method to check for accessibility violations:
+
+```js:title=cypress/e2e/a11y.test.js
+/// <reference types="Cypress" />
+
+describe("Accessibility checks", () => {
+  beforeEach(() => {
+    cy.visit("/")
+    cy.injectAxe()
+    cy.wait(500)
+  })
+  it("Has no detectable a11y violations on load", () => {
+    cy.checkA11y()
+  })
+})
+```
+
+You can run `test:e2e` to run the test. If you already have the development server open you can run `cy:open` instead.
+
+One thing to keep in mind is that you can't always see the exact error message from the sidebar (command log). For that, you have to open the developer console and find the message in the output. You can see how an accessibility error looks in the [cypress-axe GitHub page](https://github.com/avanslaars/cypress-axe#output).
+
+You don't have to use the `checkA11y` method only on page load. For example, you can perform a click on a button and check again. This is especially useful if that button opens a modal or a mobile menu for example.
+
+The following test is for the [gatsby-default-starter](https://github.com/gatsbyjs/gatsby-starter-default). You visit the homepage and you search for the link that goes to page 2 with the `getByText` command. Then you perform a click and check for accessibility errors in the second page.
```

Comment:
```suggestion
The following tests are for the [gatsby-default-starter](https://github.com/gatsbyjs/gatsby-starter-default). The first test visits the homepage and runs `cypress-axe` on it. In the second test, Cypress searches for a link that goes to page 2 with the `getByText` command, performs a click event on the link, and checks for accessibility errors on the second page.
```
The user isn't doing the work, the automated test is...so this suggestion reflects that!

Reply:
> The user isn't doing the work, the automated test is.

Yes, I agree, thanks! 

My objection here is that there is only one test, the highlighted. I'm including the previous test just to show what changed in the test file.